lib/src/utils.dart

```diff
@@ -2,23 +2,21 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-// @dart=2.9
-
 library dartdoc.utils;
 
 final RegExp leadingWhiteSpace = RegExp(r'^([ \t]*)[^ ]');
 
 Iterable<String> stripCommonWhitespace(String str) sync* {
   if (str.isEmpty) return;
   final lines = str.split('\n');
-  int /*?*/ minimumSeen;
+  int? minimumSeen;
 
   for (final line in lines) {
     if (line.isNotEmpty) {
       final match = leadingWhiteSpace.firstMatch(line);
       if (match != null) {
-        if (minimumSeen == null || match.group(1).length < minimumSeen) {
-          minimumSeen = match.group(1).length;
+        if (minimumSeen == null || match.group(1)!.length < minimumSeen) {
```

Comment:
Optional: might as well assign `match.group(1)!.length` to a local, so as to avoid the duplicate lookup, null check, and lookup.

Reply:
yes that's much better.